src/style/themes/base/base-theme.config.ts

```diff
@@ -2,15 +2,11 @@ import tokens from "@sage/design-tokens/js/base/common";
 
 import { PaletteFunction } from "../../palette";
 
-export interface ColorsWithHex {
-  [key: string]: string | ((opacity: number | string) => string);
-}
-
-export type BasePalette = Record<string, string | PaletteFunction> &
-  ColorsWithHex & {
-    blackOpacity: (opacity: number | string) => string;
-    whiteOpacity: (opacity: number | string) => string;
-  };
+export type BasePalette = Record<string, string | PaletteFunction> & {
+  blackOpacity: PaletteFunction;
+  whiteOpacity: PaletteFunction;
+  [key: string]: PaletteFunction;
```

Comment:
@samtjo @grabkowski I ended up having to do something like this as it was complaining in the Theme when I was calling the palette functions that it might be a string and therefore not callable. I tried adding `as const` etc but had no luck and some of it was bit above my understanding